### PR TITLE
fix: correctly interrupt main thread

### PR DESCRIPTION
**Pull request by timoffex**
Description
---
Fixes WB-18098, making the stop button more reliable.

Testing
---
Run the following:

```python
import time
import wandb

wandb.init()
time.sleep(9999)
```

Then press the stop button in the run's Overview tab, next to its State property. Prior to this fix, the stop button would not interrupt the sleep on any platform.

Tested manually on macOS and Windows. Tested on Linux via system-tests.

### CODE REVIEW

**Comment by kptkin** [comment]
is there a way we can test it with system tests? to avoid breaking it in the future?

**Comment by timoffex** [comment]
> is there a way we can test it with system tests? to avoid breaking it in the future?

Added!

**Comment by kptkin** [comment]
> However, I think we don't run these on Windows, do we?

we do but only for the main and release branches

**Comment by kptkin** [comment]
> However, I think we don't run these on Windows, do we?

yeah you are right we run func tests only on linux...

File: wandb/sdk/lib/interrupt.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    thread on its own.
+
+    On Windows, pthread_kill is not available and os.kill() ignores its
+    second argument and always kills the process. However,
+    signal.raise_signal() does the right thing.
+    """
+    if platform.system() == "Windows":
+        signal.raise_signal(signal.SIGINT)
+    else:
+        signal.pthread_kill(
```
Comment (kptkin) [inline wandb/sdk/lib/interrupt.py:34]: i wonder why this one is not counted against coverage

**Review by kptkin — APPROVED** [review]

File: wandb/sdk/lib/interrupt.py (hunk 1 above)
Comment (timoffex) [reply wandb/sdk/lib/interrupt.py:34]: the subprocess started by the test doesn't collect coverage

**Pull request merged**